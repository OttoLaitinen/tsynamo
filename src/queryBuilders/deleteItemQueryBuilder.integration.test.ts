import { DDB } from "../../test/testFixture";
import { getDDBClientFor, startDDBTestContainer } from "../../test/testUtil";
import { Tsynamo } from "./../index";

describe("DeleteItemQueryBuilder", () => {
  let tsynamoClient: Tsynamo<DDB>;

  beforeAll(async () => {
    const testContainer = await startDDBTestContainer();

    tsynamoClient = new Tsynamo<DDB>({
      ddbClient: await getDDBClientFor(testContainer),
    });
  });

  it("handles a simple delete query", async () => {
    await tsynamoClient
      .putItem("myTable")
      .item({
        userId: "1",
        dataTimestamp: 2,
      })
      .execute();

    const itemBeforeDeletion = await tsynamoClient
      .getItemFrom("myTable")
      .keys({ userId: "1", dataTimestamp: 2 })
      .execute();

    expect(itemBeforeDeletion).toBeDefined();

    const deleteResponse = await tsynamoClient
      .deleteItem("myTable")
      .keys({
        userId: "1",
        dataTimestamp: 2,
      })
      .returnValues("ALL_OLD")
      .execute();

    expect(deleteResponse).toMatchSnapshot();

    const itemAfterDeletion = await tsynamoClient
      .getItemFrom("myTable")
      .keys({ userId: "1", dataTimestamp: 2 })
      .execute();

    expect(itemAfterDeletion).toBeUndefined();
  });
});
