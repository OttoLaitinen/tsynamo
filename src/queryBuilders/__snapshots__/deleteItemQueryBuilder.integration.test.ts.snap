// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DeleteItemQueryBuilder > handles a simple delete query 1`] = `
{
  "dataTimestamp": 2,
  "userId": "1",
}
`;
